libs/environment/src/components/network-switcher/network-switcher.tsx

```diff
@@ -0,0 +1,177 @@
+import classNames from 'classnames';
+import { Fragment, useState, useCallback } from 'react';
+import * as DropdownMenuPrimitive from '@radix-ui/react-dropdown-menu';
+import { t } from '@vegaprotocol/react-helpers';
+import {
+  Link,
+  Icon,
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+} from '@vegaprotocol/ui-toolkit';
+import { useEnvironment } from '../../hooks/use-environment';
+import { Networks } from '../../types';
+
+export const envNameMapping: Record<Networks, string> = {
+  [Networks.CUSTOM]: t('Custom'),
+  [Networks.DEVNET]: t('Devnet'),
+  [Networks.STAGNET]: t('Stagnet'),
+  [Networks.STAGNET2]: t('Stagnet2'),
+  [Networks.STAGNET3]: t('Stagnet3'),
+  [Networks.TESTNET]: t('Fairground testnet'),
+  [Networks.MAINNET]: t('Mainnet'),
+};
+
+export const envTriggerMapping: Record<Networks, string> = {
+  ...envNameMapping,
+  [Networks.TESTNET]: t('Fairground'),
+};
+
+export const envDescriptionMapping: Record<Networks, string> = {
+  [Networks.CUSTOM]: '',
+  [Networks.DEVNET]: t('The latest Vega code auto-deployed'),
+  [Networks.STAGNET]: t('A staging environment with trading'),
+  [Networks.STAGNET2]: t('A staging envirnment without trading'),
+  [Networks.STAGNET3]: t(
+    'A testnet that simulates validators coming and going'
+  ),
+  [Networks.TESTNET]: t(
+    'Public testnet run by the Vega team, often used for incentives'
+  ),
+  [Networks.MAINNET]: t('The vega mainnet'),
+};
+
+const standardNetworkKeys = [Networks.MAINNET, Networks.TESTNET];
+const advancedNetworkKeys = [
+  Networks.MAINNET,
+  Networks.TESTNET,
+  Networks.STAGNET3,
+  Networks.STAGNET2,
+  Networks.STAGNET,
+  Networks.DEVNET,
+];
+
+type NetworkLabelProps = {
+  isCurrent: boolean;
+  isAvailable: boolean;
+};
+
+const NetworkLabel = ({
+  isCurrent = false,
+  isAvailable = false,
+}: NetworkLabelProps) => (
+  <span className="text-white-80 dark:text-white-80">
+    {isCurrent
+      ? ` (${t('current')})`
+      : !isAvailable
+      ? ` (${t('not available')})`
+      : ''}
+  </span>
+);
+
+export const NetworkSwitcher = () => {
+  const { VEGA_ENV, VEGA_NETWORKS } = useEnvironment();
+  const [isOpen, setOpen] = useState(false);
+  const [isAdvancedView, setAdvancedView] = useState(false);
+
+  const handleOpen = useCallback(
+    (isOpen: boolean) => {
+      setOpen(isOpen);
+      if (!isOpen) {
+        setAdvancedView(false);
+      }
+    },
+    [setOpen, setAdvancedView]
+  );
+
+  const menuItemClasses = 'pt-12 pb-12 pl-16 pr-16 h-auto';
```

Comment:
```suggestion
  const menuItemClasses = 'py-12 px-16 h-auto';
```

Reply:
this doesn't work, as the dropdown classes override them, that's why i had to hack it in this way. or i could also potentially just not use the dropdown item